transports/http/handlers/webhooks_handler.go

```diff
@@ -0,0 +1,93 @@
+package handler
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+	webhook "github.com/libsv/bitcoin-hc/transports/http"
+)
+
+// nolint: godot
+// registerWebhook godoc.
+//
+//	@Summary Register new webhook
+//	@Tags webhooks
+//	@Accept json
+//	@Produce json
+//	@Success 200 {object} domains.Webhook
+//	@Router /webhook [post]
+//	@Param data body http.WebhookRequest true "Webhook to register"
+//
+// @Security Bearer
+func (h *Handler) registerWebhook(c *gin.Context) {
+	var reqBody webhook.WebhookRequest
+	err := c.Bind(&reqBody)
+
+	fmt.Println("WEBHOOK: ", reqBody)
+
+	if err == nil {
+
+		if reqBody.Url == "" {
+			c.JSON(http.StatusBadRequest, "Url is required")
+			return
+		}
+		var tHeader, token string
+
+		// If custom header is specified, use it, otherwise use default
+		if strings.ToLower(reqBody.RequiredAuth.Type) == "custom_header" {
+			tHeader = reqBody.RequiredAuth.Header
+			token = reqBody.RequiredAuth.Token
+		} else {
+			tHeader = "Authorization"
+			token = "Bearer " + reqBody.RequiredAuth.Token
+		}
```

Comment:
IMHO that is a part of the logic of creating a webhook, so it rather should be placed in a service.
I think also that validation could be part of the create method in service.